weaviate/collections/batch/batch_wrapper.py

```diff
@@ -26,26 +27,19 @@ def __init__(
         self._concurrent_requests: int = 2
 
         self._batch_data = _BatchDataWrapper()
-        self.__shut_background_thread_down: Optional[threading.Event] = None
 
     def __start_event_loop_thread(self, loop: asyncio.AbstractEventLoop) -> None:
-        while (
-            self.__shut_background_thread_down is not None
-            and not self.__shut_background_thread_down.is_set()
-        ):
-            if loop.is_running():
-                continue
-            else:
-                loop.run_forever()
-
-    def _open_async_connection(self) -> asyncio.AbstractEventLoop:
+        loop.set_debug(True)  # in case of errors, shows async errors in the thread to users
+        loop.run_forever()
```

Comment:
I think I had crashes here, when the eventloop was already running for whatever reason

Reply:
Yeh, this is what `nest_asyncio` fixes. By patching the loop, we're allowed to call `run_forever` on it again and therefore take the global loop from the main thread and put it into our background event loop. There's no longer a need to loop through indefinitely since `loop.run_forever` now blocks until `loop.stop` is called in `__exit__`

Reply:
Have removed `nest_asyncio` in favour of creating a new event loop in all cases. Now `run_forever` always works